### PR TITLE
feat: allow fields remapping in JSON logs

### PR DESCRIPTION
**Pull request by redbaron**
First stab at #750

### CODE REVIEW

File: pkg/management/log/flags.go [hunk 1]
```diff
@@ -104,6 +116,44 @@ func getLogLevelString(l zapcore.Level) string {
...
+			if cfg.TimeKey != "" {
+				c.TimeKey = cfg.TimeKey
+			}
+			if cfg.LevelKey != "" {
+				c.LevelKey = cfg.LevelKey
+			}
+		})
+	}
+}
+
```
[the remark this answers is not in the document]
Comment (redbaron) [reply pkg/management/log/flags.go]: created two separate CLI flags. Stackdriver compatible output is then can be set with:

```
--zap-encoder=json --log-field-level=severity --log-field-timestamp=time --zap-time-encoding rfc3339nano
```

[the remark this answers is not in the document]
Comment (gbartolini) [reply pkg/management/log/flags.go]: I agree with @phisco - sounds a good idea

**Comment by gbartolini** [comment]
@redbaron - would it be possible to document how this customization can happen? Thanks!

**Comment by phisco** [comment]
We discussed it during today's bi-weekly meeting and agreed we could move this forward and document this by adding it as a parameter to the cnpg subcommand generating installation manifests and the helm chart.

**Comment by gbartolini** [comment]
One idea we discussed in the dev meeting is to add those two options to the `kubectl cnpg install generate` command, and document this in:

- [the plugin page](https://cloudnative-pg.io/documentation/current/cnpg-plugin/#generation-of-installation-manifests)
- [the logging page](https://cloudnative-pg.io/documentation/current/logging/) - when we explain the format we could add a box that mentions that those two fields can be renamed

**Comment by redbaron** [comment]
> One idea we discussed in the dev meeting is to add those two options to the `kubectl cnpg install generate` command, and document this in:
> 
>     * [the plugin page](https://cloudnative-pg.io/documentation/current/cnpg-plugin/#generation-of-installation-manifests)
> 
>     * [the logging page](https://cloudnative-pg.io/documentation/current/logging/) - when we explain the format we could add a box that mentions that those two fields can be renamed


I've added box to the logging page. As for the generate command, these new flags are no more special than many existing flags. Generate command produces yaml, which system administrators can trivially edit to add more flags.

File: docs/src/logging.md [hunk 2]
```diff
@@ -20,6 +20,10 @@ Each log entry has the following fields:
     [Kubernetes Logging Architecture](https://kubernetes.io/docs/concepts/cluster-administration/logging/)
     documentation.
 
+!!! Info
+    `level` and `ts` fields can be renamed with `log-field-level` and `log-field-timestamp`
+    CLI flags should you log ingestion system require it.
+
```
Comment (gbartolini) [inline docs/src/logging.md:26]:
```suggestion
!!! Info
    The `level` and `ts` field names can be renamed via the `log-field-level` and
    `log-field-timestamp` flags of the operator controller, should your log ingestion
    system require it. All you have to do is edit the `Deployment` definition of the
    `cloudnative-pg` operator.

```

**Review by gbartolini — APPROVED** [review]

**Review by phisco — DISMISSED** [review]

**Comment by phisco** [comment]
I realized the patch is not actually passing the mapping down to the cluster's pods, shouldn't be too hard, I'll try addressing that.

**Review by phisco — APPROVED** [review]

**Comment by phisco** [comment]
I switched defaults to empty string to be able to differentiate between default values and configured values to avoid triggering a restart of all running clusters upon upgrade.

**Comment by redbaron** [comment]
Thanks for pushing it!

**Pull request merged**